cloud/kubernetes: update manifests and enhance security

shreyaskm623 opened the pull request:
This commit updates the Kubernetes deployment manifests by removing deprecated APIs, enhancing security with IMDSv2, improving ARM compatibility, and cleaning up legacy files.

Key changes:
- Update PodDisruptionBudget API from policy/v1beta1 to policy/v1
- Replace byrnedo/alpine-curl:0.1 with 3.20 for ARM64 support
- Implement IMDSv2 for secure AWS metadata access in EKS deployments
- Remove deprecated v1.6 and v1.7 directories (22 legacy files)
- Regenerate all manifest files from updated templates

Release notes: none

Epic: CRDB-49252

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/151314)
<!-- Reviewable:end -->

shreyaskm623 committed:
cloud/kubernetes: update manifests and enhance security

  This commit updates the Kubernetes deployment manifests by removing deprecated APIs, enhancing security with IMDSv2, improving ARM compatibility, and cleaning up legacy files.

  Key changes:

  Update PodDisruptionBudget API from policy/v1beta1 to policy/v1
  Replace byrnedo/alpine-curl:0.1 with 3.20 for ARM64 support
  Implement IMDSv2 for secure AWS metadata access in EKS deployments
  Remove deprecated v1.6 and v1.7 directories (22 legacy files)
  Regenerate all manifest files from updated templates

  Release notes: none

  Epic: CRDB-49252

NishanthNalluri commented on cloud/kubernetes/templates/v1.20/cockroachdb-statefulset-secure.yaml:
```diff
@@ -191,11 +196,38 @@ spec:
       - name: cockroachdb
         image: cockroachdb/cockroach:@VERSION@
         imagePullPolicy: IfNotPresent
+        # TODO: Change these to appropriate values for the hardware that you're running. You can see
```
Nit: I suppose we do not need to add `TODO` here?

NishanthNalluri commented on cloud/kubernetes/v1.20/cockroachdb-statefulset-secure.yaml:
```diff
@@ -190,13 +195,40 @@ spec:
       containers:
       - name: cockroachdb
-        image: cockroachdb/cockroach:v25.2.4
+        image: cockroachdb/cockroach:@VERSION@
```
Is `@VERSION@` a placeholder?

shreyaskm623 committed:
Merge branch 'master' of github.com:cockroachdb/cockroach into k8s_manifests_refactor
